### PR TITLE
Introduce kludge to make "isEnabledForCluster" work again for cluster page menus  PART 2

### PR DESCRIPTION
**Pull request by jansav**
Part 2 of #5800.

Signed-off-by: Janne Savolainen <janne.savolainen@live.fi>

### CODE REVIEW

**Review by Nokel81 — APPROVED** [review]

File: src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import asyncFn from "@async-fn/jest";
+import type { RenderResult } from "@testing-library/react";
+import type { ApplicationBuilder } from "../../../renderer/components/test-utils/get-application-builder";
+import { getApplicationBuilder } from "../../../renderer/components/test-utils/get-application-builder";
+import { getExtensionFakeFor } from "../../../renderer/components/test-utils/get-extension-fake";
+import type { KubernetesCluster } from "../../../common/catalog-entities";
+import React from "react";
+import extensionShouldBeEnabledForClusterFrameInjectable from "../../../renderer/extension-loader/extension-should-be-enabled-for-cluster-frame.injectable";
+
+describe("disable-sidebar-items-when-cluster-is-not-relevant", () => {
```
Comment (Nokel81) [inline src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx]: nit: test names shouldn't use kebab-case

Comment (jim-docker) [reply src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx]: it's meant to match the test file name?

File: src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        ],
+      },
+    });
+
+    rendered = await builder.render();
+
+    builder.extensions.enable(testExtension);
+  });
+
+  describe("given not yet known if extension should be enabled for the cluster, when navigating", () => {
```
Comment (jim-docker) [inline src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx]: Is there navigation involved?

**Review by jim-docker — APPROVED** [review]

File: src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx (hunk 2 above)
Comment (jansav) [reply src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx]: Good catch.

File: src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx (hunk 1 above)
Comment (jansav) [reply src/behaviours/cluster/extension-api/disable-sidebar-items-when-cluster-is-not-relevant.test.tsx]: Changed.

**Pull request merged**